Common/Model.cs

```diff
@@ -0,0 +1,143 @@
+            Scene scene = importer.ImportFile(path, PostProcessSteps.Triangulate);
+
+            // check for errors
+            if (scene == null || scene.SceneFlags.HasFlag(SceneFlags.Incomplete) || scene.RootNode == null)
+            {
+                Console.WriteLine("Unable to load model from: " + path);
+                return;
+            }
+            
+            //Reset the meshes and textures
+            meshes = new List<Mesh>();
+
+            //Set the scale of the model
+            float scale = 1/200.0f;
+            Matrix4x4 scalingMatrix = new Matrix4x4(scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, 1);
+
+            // process ASSIMP's root node recursively. We pass in the scaling matrix as the first transform
+            ProcessNode(scene.RootNode, scene, scalingMatrix);
+
+            importer.Dispose();
+        }
+
+        // processes a node in a recursive fashion. Processes each individual mesh located at the node and repeats this process on its children nodes (if any).
+        private void ProcessNode(Node node, Scene scene, Matrix4x4 parentTransform)
+        {
+            //Multiply the transform of each node by the node of the parent, this will place the meshes in the correct relative location
+            Matrix4x4 transform = node.Transform*parentTransform;
+
+            // process each mesh located at the current node
+            for (int i = 0; i < node.MeshCount; i++)
+            {
+                // the node object only contains indices to index the actual objects in the scene. 
+                // the scene contains all the data, node is just to keep stuff organized (like relations between nodes).
+                AssimpMesh mesh = scene.Meshes[node.MeshIndices[i]];
+                meshes.Add(ProcessMesh(mesh, transform));
+            }
+
+            for (int i = 0; i < node.ChildCount; i++)
+            {
+                ProcessNode(node.Children[i], scene, transform);
+            }
+        }
+
+        private Mesh ProcessMesh(AssimpMesh mesh, Matrix4x4 transform)
+        {
+            // data to fill
+            List<Vertex> vertices = new List<Vertex>();
+            List<int> indices = new List<int>();
+
+            // walk through each of the mesh's vertices
+            for (int i = 0; i < mesh.VertexCount; i++)
+            {
+                Vertex vertex = new Vertex();
+                // we declare a placeholder vector since assimp uses its own vector class that doesn't directly convert to the Vector3 class so we transfer the data to this placeholder Vector3 first.
+
+                // positions
+                var transformedVertex = transform * mesh.Vertices[i];
+                Vector3 vector;
+                vector.X = transformedVertex.X;
+                vector.Y = transformedVertex.Y;
+                vector.Z = transformedVertex.Z;
```

Comment:
We should do this more efficiently, it's possible to use an extension method that does a reinterpret cast of the vector so there is no need to copy the fields individually. But also, we should just convert the `System.Numerics` type to an OpenTK vector before multiplying with the matrix. 

I can help with this code if you want.

Reply:
I gave it a shot using extension methods to convert Vectors and Matrices from Assimp to OpenTk, but I wasn't sure how to go about just converting the Assimp types into System.Numerics types like you mention here.

Reply:
I was confused and thought Assimp used `System.Numerics`. Your extension methods where basically what I had in mind. I changed it to use a "reinterpret cast", though as we are passing the vector by value I'm not sure if this actually makes a performance difference. 

I could be convinced to change it back if the code is confusing.

Reply:
I like the way that you have done it. Would it be possible to pass a reference to the extension methods to do the reinterpret cast so that we aren't passing by value?